libraries/fbx/src/FBXReader_Mesh.cpp

```diff
@@ -422,8 +422,18 @@ void FBXReader::buildModelMesh(FBXMesh& extractedMesh, const QString& url) {
     int colorsSize = fbxMesh.colors.size() * sizeof(glm::vec3);
     int texCoordsSize = fbxMesh.texCoords.size() * sizeof(glm::vec2);
     int texCoords1Size = fbxMesh.texCoords1.size() * sizeof(glm::vec2);
-    int clusterIndicesSize = fbxMesh.clusterIndices.size() * sizeof(glm::vec4);
-    int clusterWeightsSize = fbxMesh.clusterWeights.size() * sizeof(glm::vec4);
+#define UNFORTUNATE_HACK
+#ifdef UNFORTUNATE_HACK
+    // HACK: store clusterIndices in floats rather than short integers
+    int clusterIndicesSize = fbxMesh.clusterIndices.size() * sizeof(float);
+#else // UNFORTUNATE_HACK
+    int clusterIndicesSize = fbxMesh.clusterIndices.size() * sizeof(uint8_t);
+    if (fbxMesh.clusters.size() > UINT8_MAX) {
+        // we need 16 bits instead of just 8 for clusterIndices
+        clusterIndicesSize *= 2;
+    }
+#endif // UNFORTUNATE_HACK
+    int clusterWeightsSize = fbxMesh.clusterWeights.size() * sizeof(uint8_t);
```

Comment:
I'm able to send the `clusterWeights` as "normalized" uint8's but when I try to use shorter data formats for the `clusterIndices` the rendering breaks.

The `clusterIndices` are definitely correct because in the `#ifdef UNFORTUNATE_HACK` sections I copy them to floats and submit to the render pipeline in the old float format and everything works fine.  Comment out the `#define UNFORTUNATE_HACK` to submit the data as shorter ints.